### PR TITLE
Ignore value conflicts when reencrypting secrets

### PR DESCRIPTION
**Pull request by dereknola**
Signed-off-by: Derek Nola <derek.nola@suse.com>

#### Proposed Changes ####
- Ignore conflicts when reencyrpting secrets. This means that the secrets has already been modified, and thus has  been reencrypted with the new key outside of the dedicated reencryption process. 

#### Types of Changes ####
Secrets Encryption Bugfix

#### Verification ####
- Deploy 1K secrets and go thru the standard secrets encryption process. See https://github.com/k3s-io/k3s/issues/5933
- While reencryption is running, pick a "high number" secret and modify its value. 
- Reencryption should complete without complaint

#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/6824

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

**Review by brandond — APPROVED** [review]

File: pkg/secretsencrypt/controller.go [hunk 1]
```diff
@@ -223,9 +223,14 @@ func (h *handler) updateSecrets(node *corev1.Node) error {
 	i := 0
 	err = meta.EachListItem(secretsList, func(obj runtime.Object) error {
 		if secret, ok := obj.(*corev1.Secret); ok {
-			if _, err := h.secrets.Update(secret); err != nil {
-				return fmt.Errorf("failed to reencrypted secret: %v", err)
-			}
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				secret, err = h.secrets.Get(secret.Namespace, secret.Name, metav1.GetOptions{})
+				if err != nil {
```
Comment (brandond) [inline pkg/secretsencrypt/controller.go:228]: What if the secret was deleted? Feels like we should ignore NotFound errors. I'm also having second thoughts about retrying on conflict; if there's a conflict it means that the secret has been updated by something else (and reencrypted in doing so) so we probably don't need to do anything and can just continue on to the next secret.

**Comment by dereknola** [comment]
@brandond your logic makes sense. I have changed the approach to this PR.

File: pkg/secretsencrypt/controller.go [hunk 2]
```diff
@@ -223,7 +224,7 @@ func (h *handler) updateSecrets(node *corev1.Node) error {
 	i := 0
 	err = meta.EachListItem(secretsList, func(obj runtime.Object) error {
 		if secret, ok := obj.(*corev1.Secret); ok {
-			if _, err := h.secrets.Update(secret); err != nil {
+			if _, err := h.secrets.Update(secret); err != nil && !apierrors.IsConflict(err) {
 				return fmt.Errorf("failed to reencrypted secret: %v", err)
```
Comment (brandond) [inline pkg/secretsencrypt/controller.go:228]:
```suggestion
				return fmt.Errorf("failed to update secret: %v", err)
```

**Review by brandond — APPROVED** [review]
LGTM, but maybe also fix the weird grammar on the error while we're here?

**Review by VestigeJ — APPROVED** [review]
LGTM

**Pull request merged**